scripts/functions.py

```diff
@@ -1,27 +1,74 @@
 #####################################
-## This file is for functions common between multiple files
+## This file is for functions and constants common between multiple files
 #####################################
 
 import datetime
 import pandas as pd
 from collections import defaultdict
+import yaml
+
+#####################################
+# NON-CONFIGURABLE CONSTANTS
+#####################################
+
+# Province to Time Zone Mapping
+PROVINCIAL_TIME_ZONES = {
+    'BC':0,
+    'AB':1,
+    'SAS':2,
+    'MAN':2,
+    'ONT':3,
+    'QC':3,
+    'NB':4,
+    'NL':4,
+    'NS':4,
+    'PEI':4}
+
+# Residual Hydro Capacity (GW) per province in 2017
+# Source: https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=2510002201&pickMembers%5B0%5D=1.1&pickMembers%5B1%5D=2.1&cubeTimeFrame.startYear=2017&cubeTimeFrame.endYear=2017&referencePeriods=20170101%2C20170101
+RESIDUAL_HYDRO_CAPACITY = {
+    'BC':  15.407,
+    'AB':  1.218,   
+    'SAS': 0.867,   
+    'MAN': 5.461,   
+    'ONT': 9.122,   
+    'QC':  40.438,  
+    'NB':  0.968,   
+    'NL':  6.762,   
+    'NS':  0.370,   
+    'PEI': 0.000}
+
+# Hydro generation (TWh) per province in 2017
+# Source: https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=2510001501&pickMembers%5B0%5D=1.1&pickMembers%5B1%5D=2.1&cubeTimeFrame.startMonth=01&cubeTimeFrame.startYear=2017&cubeTimeFrame.endMonth=12&cubeTimeFrame.endYear=2017&referencePeriods=20170101%2C20171201
+HYDRO_GENERATION = {
+    'BC': 66.510,
+    'AB': 2.050,
+    'SAS': 3.862,
+    'MAN': 35.991,
+    'ONT': 39.492,
+    'QC': 202.001,
+    'NB': 2.583,
+    'NL': 36.715,
+    'NS': 0.849,
+    'PEI': 0.000}
+
+#####################################
+# COMMON FUNCTIONS
+#####################################
+def openYaml():
+    # PURPOSE: Returns the configured config.yaml file, which is a modifiable settings file that contains constants
+    # INPUT: None
+    # OUTPUT: parsedYaml = the ready-to-use yaml file
+
+    originalYaml = open("../scripts/config.yaml")

```

Comment:
The yaml file is associated with the scripts, not the osemosys source, so I would leave it as is, since the scripts are not currently part of "the source". I've brought this up once before, I think, but imho the entire scripts folder should be moved into the src folder, in which case the yaml file would come along.

Reply:
Yeah thats fair. I guess where Im coming from is trying to keep anything the user should interact with in one folder, and anything they shouldn't totally separate.

Next time we chat in person lets discuss this